src/explorer/setAppWizardContextDefault.ts

```diff
@@ -9,32 +9,37 @@ import { Uri, workspace, WorkspaceConfiguration } from 'vscode';
 import { LocationListStep } from 'vscode-azureextensionui';
 import { configurationSettings, extensionPrefix } from '../constants';
+import { IDeployWizardContext } from '../IDeployWizardContext';
 import { javaUtils } from '../utils/javaUtils';
 
-export async function setAppWizardContextDefault(wizardContext: IAppServiceWizardContext): Promise<void> {
-    const isJavaProject: boolean = await javaUtils.isJavaProject();
+export async function setAppWizardContextDefault(wizardContext: IAppServiceWizardContext & IDeployWizardContext): Promise<void> {
+    // if the user entered through "Deploy", we'll have a project to base our recommendations on
+    // otherwise, look at their current workspace and only suggest if one workspace is opened
+    const workspaceForRecommendation: string | undefined = wizardContext.fsPath ?
+        wizardContext.fsPath : workspace.workspaceFolders && workspace.workspaceFolders.length === 1 ?
+            workspace.workspaceFolders[0].uri.fsPath : undefined;
 
-    if (isJavaProject) {
-        wizardContext.recommendedSiteRuntime = [
-            LinuxRuntimes.java,
-            LinuxRuntimes.tomcat,
-            LinuxRuntimes.wildfly
-        ];
+    if (workspaceForRecommendation) {
+        const fsPath: string = workspaceForRecommendation;
 
-        // considering high resource requirement for Java applications, a higher plan sku is set here
-        wizardContext.newPlanSku = { name: 'P1v2', tier: 'PremiumV2', size: 'P1v2', family: 'P', capacity: 1 };
-        // to avoid 'Requested features are not supported in region' error
-        await LocationListStep.setLocation(wizardContext, 'weseteurope');
-    }
-
-    // only detect if one workspace is opened
-    if (workspace.workspaceFolders && workspace.workspaceFolders.length === 1) {
-        const fsPath: string = workspace.workspaceFolders[0].uri.fsPath;
         if (await fse.pathExists(path.join(fsPath, 'package.json'))) {
             wizardContext.recommendedSiteRuntime = [LinuxRuntimes.node];
+
         } else if (await fse.pathExists(path.join(fsPath, 'requirements.txt'))) {
             // requirements.txt are used to pip install so a good way to determine it's a Python app
             wizardContext.recommendedSiteRuntime = [LinuxRuntimes.python];
+
+        } else if (await javaUtils.isJavaProject(fsPath)) {
+            wizardContext.recommendedSiteRuntime = [
+                LinuxRuntimes.java,
+                LinuxRuntimes.tomcat,
+                LinuxRuntimes.wildfly
+            ];
+
+            // considering high resource requirement for Java applications, a higher plan sku is set here
+            wizardContext.newPlanSku = { name: 'P1v2', tier: 'PremiumV2', size: 'P1v2', family: 'P', capacity: 1 };
+            // to avoid 'Requested features are not supported in region' error
+            await LocationListStep.setLocation(wizardContext, 'weseteurope');
```

Comment:
For L62-65 where we recommend Windows based on a `csproj` file, there wasn't a clean way to check the current fsPath since I am using `workspace.findFiles`.  I can use fse and recursively search their directory but wasn't sure if it was worth it

Reply:
findFiles lets you pass in a `RelativePattern` to restrict the glob to just a single workspace

Reply:
The `RelativePattern` only works if the workspace is opened so if the user deployed a not opened workspace, findFiles won't recognize that fsPath. :/

Reply:
Never mind, I'm wrong about this.  I didn't understand how to use `RelativePattern`s